src/core/services/shops.service.ts

```diff
@@ -1,67 +1,134 @@
-import { Injectable } from '@angular/core';
-import { HttpClient, HttpHeaders } from '@angular/common/http';
-import { LatLng, UserLogin, Category } from '../models/Models';
-import { Observable } from 'rxjs';
-import { Shop } from '../models/shop.model';
-import { config } from '../../configs/baseconfig'
+import {Injectable} from '@angular/core';
+import {HttpClient, HttpHeaders} from '@angular/common/http';
+import {LatLng, UserLogin, Category} from '../models/Models';
+import {Observable} from 'rxjs';
+import {Shop} from '../models/shop.model';
+import {config} from '../../configs/baseconfig';
 
 @Injectable({
-  providedIn: 'root'
+  providedIn: 'root',
 })
 export class ShopsService {
-
   private hyderabadLatng: LatLng = {
-    latitude: 17.3850,
-    longitude: 78.4867
-  }
+    latitude: 17.385,
+    longitude: 78.4867,
+  };
+
+  constructor(private http: HttpClient) {}
 
-  constructor(private http: HttpClient) { }
+  addShop(shopObject): Observable<any> {
+    const userDetails: any = JSON.parse(localStorage.getItem('shopsnearbyme'));
+    let body = new FormData();
+    body.set('name', shopObject.shopName);
+    body.set('category', shopObject.shopCategory);
+    body.set('description', shopObject.shopDescription);
+    body.set('phoneNumber', userDetails.phoneNumber);
+    body.set('timeFrom', shopObject.startTime);
+    body.set('timeTo', shopObject.endTime);
+    body.set('address', shopObject.shopAddress);
+    body.set('latitude', userDetails.latitude);
+    body.set('longitude', userDetails.longitude);
+    if (shopObject.imageFile) {
+      body.set('image', shopObject.imageFile);
+    }
+    let options = {
+      headers: this.setBasicAuthHeader({
+        email: userDetails.email,
+        password: userDetails.password,
+      }),
+    };
+    return this.http.post<any>(`${config.API_URL}/shops/others`, body, options);
+  }
 
+  getNearByDiscountsAndCoupons() {
+    const userDetails: any = JSON.parse(localStorage.getItem('shopsnearbyme'));
+    const latlng = this.sanitizeLatLng({
+      latitude: userDetails.latitude,
+      longitude: userDetails.longitude,
+    });
+    let url =
+      config.API_URL +
+      '/shops/discountsAndCoupons?latitude=' +
+      latlng.latitude +
+      '&longitude=' +
+      latlng.longitude +
+      '&range=20000';
+    return this.simpleGetRequest<any>(url, {
+      email: userDetails.email,
+      password: userDetails.password,
+    });
+  }
 
   public getRecommendedAndOtherShops(
-    userlogin: UserLogin,
-    latlng: LatLng,
     categories: Category[],
-    distance: number): Observable<Shop[]> {
-    latlng = this.sanitizeLatLng(latlng)
-    let url: string = config.API_URL + "/shops/recommendedAndOthers?latitude=" + latlng.latitude
-      + "&longitude=" + latlng.longitude + "&range=" + distance;
+    distance: number
+  ): Observable<any> {
+    const userDetails: any = JSON.parse(localStorage.getItem('shopsnearbyme'));
```

Comment:
Can we make some modification to make userDetails of type UserLogin instead of any?
Shop service should not know anything about how Authentication details are fetched. 
Authentication details should come to shops service as a parameter.